### PR TITLE
dealii: 9.2.0 release.

### PR DESCRIPTION
**Pull request by luca-heltai**
Version 9.2.0 of the library just came out. Finer tuning may be possible, but this already works on my machine.

### CODE REVIEW

File: var/spack/repos/builtin/packages/dealii/package.py [hunk 1]
```diff
@@ -23,6 +23,7 @@ class Dealii(CMakePackage, CudaPackage):
     transitive_rpaths = False
 
     version('develop', branch='master')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/dealii/package.py]:
```suggestion
    version('master', branch='master')
```

**Comment by adamjstewart** [comment]
@davydden @jppelteret can you review this?

File: var/spack/repos/builtin/packages/dealii/package.py (hunk 1 above)
Comment (jppelteret) [reply var/spack/repos/builtin/packages/dealii/package.py]: Is this necessary? Will I now have to write `spack install dealii@master`, or will `spack install dealii@develop` still work?

**Review by jppelteret — APPROVED** [review]

File: var/spack/repos/builtin/packages/dealii/package.py (hunk 1 above)
Comment (jppelteret) [reply var/spack/repos/builtin/packages/dealii/package.py]: Maybe let me rephrase what I wrote above: Is this change being implemented throughout spack?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/dealii/package.py]: Once upon a time, master was considered older than all stable releases. For this reason, a lot of packages were forced to rename master to develop for things to be handled correctly. This has since been fixed, and master is treated similar to develop, so the rename is no longer necessary. I’ve been slowly trying to update packages across Spack to call master 'master'. You would have to write `dealii@master`.

**Review by alalazo — APPROVED** [review]

**Review by davydden — APPROVED** [review]
LGTM apart from renaming

**Review by adamjstewart — APPROVED** [review]
